### PR TITLE
Add binaries built for multiple windows platforms

### LINKED ISSUE #601: net461 version of dotnet-aspnet-codegenerator-design.exe should be built for multiple platforms instead of win7-x86
For a net461 project with RID as win7-x64, the dotnet-aspnet-codegenerator-design.exe causes exception as MVC tries to load the 32-bit assembly in a 64-bit process:

```
System.BadImageFormatException occurred
  HResult=0x8007000B
  Message=Could not load file or assembly 'dotnet-aspnet-codegenerator-design' or one of its dependencies. An attempt was made to load a program with an incorrect format.
  Source=mscorlib 
```

Sample project: 

```xml
<Project Sdk="Microsoft.NET.Sdk.Web">

  <PropertyGroup>
    <TargetFramework>net461</TargetFramework>
    <RuntimeIdentifier>win7-x64</RuntimeIdentifier>
  </PropertyGroup>

  <ItemGroup>
    <Folder Include="wwwroot\" />
  </ItemGroup>

  <ItemGroup>
    <PackageReference Include="Microsoft.AspNetCore" Version="2.0.0" />
    <PackageReference Include="Microsoft.AspNetCore.Mvc" Version="2.0.0" />
    <PackageReference Include="Microsoft.AspNetCore.Mvc.Formatters.Xml" Version="2.0.0" />
    <PackageReference Include="Microsoft.VisualStudio.Web.CodeGeneration.Design" Version="2.0.0" />
  </ItemGroup>

  <ItemGroup>
    <DotNetCliToolReference Include="Microsoft.VisualStudio.Web.CodeGeneration.Tools" Version="2.0.0" />
  </ItemGroup>

</Project>
```

### PR DESCRIPTION
**Pull request by prafullbhosale**
#### Changes include: 
- Add binaries for the following RIDs to the Microsoft.VisualStudio.Web.CodeGeneration.Design package
  - win7-x86
  - win7-x64
  - win-arm
  - win-arm64
- Also add a `AnyCPU` PlatformTarget binary in `lib\net461` path to handle the case when the user project specifies `<PlatformTarget>AnyCpu</PlatformTarget>` without a RID

- Change how we locate the _inside-man_ to dispatch for net461 case to handle the fact that the compilation assemblies will no longer have the _inside man_ for net461 case

#### Impact of packing 6 binaries instead of 2: 
On local machine the `2.0.0` package size is 55KB, the new package size is 127 KB

cc @Eilon, @mlorbetske 

Fixes #601

### CODE REVIEW

File: src/Shared/DotNetDispatcher/DotnetToolDispatcher.cs [hunk 1]
```diff
@@ -71,7 +71,7 @@ internal static class DotnetToolDispatcher
             {
                 EnsureBindingRedirects(assemblyFullPath, Path.GetFileName(dependencyToolPath));
                 // For Full framework, we can directly invoke the <dependencyTool>.exe from the user's bin folder.
-                dependencyToolPath = Path.Combine(Path.GetDirectoryName(assemblyFullPath), Path.GetFileName(dependencyToolPath));
+                // dependencyToolPath = Path.Combine(Path.GetDirectoryName(assemblyFullPath), Path.GetFileName(dependencyToolPath));
```
Comment (pranavkm) [inline src/Shared/DotNetDispatcher/DotnetToolDispatcher.cs]: Delete?

File: src/Microsoft.VisualStudio.Web.CodeGeneration.Tools/Program.cs [hunk 2]
```diff
@@ -193,6 +205,15 @@ private static void Execute(string[] args, bool isNoBuild, ILogger logger)
                     assemblyFullPath: context.AssemblyFullPath);
         }
 
+        private static bool IsNetCoreAppFramework(NuGetFramework framework)
+        {
+            //Only need to compare the framework name to be netcoreapp. Version doesn't matter.
```
Comment (pranavkm) [inline src/Microsoft.VisualStudio.Web.CodeGeneration.Tools/Program.cs]: `// Only` (spacing)

File: src/Microsoft.VisualStudio.Web.CodeGeneration.Tools/Program.cs [hunk 3]
```diff
@@ -166,15 +166,27 @@ private static void Execute(string[] args, bool isNoBuild, ILogger logger)
...
-            if (string.IsNullOrEmpty(dotnetCodeGenInsideManPath))
+                if (string.IsNullOrEmpty(dotnetCodeGenInsideManPath))
+                {
+                    throw new InvalidOperationException(Resources.AddDesignPackage);
+                }
+            }
+            else
             {
-                throw new InvalidOperationException(Resources.AddDesignPackage);
+                dotnetCodeGenInsideManPath = Path.Combine(Path.GetDirectoryName(context.AssemblyFullPath), DESIGN_TOOL_NAME+".exe");
```
Comment (pranavkm) [inline src/Microsoft.VisualStudio.Web.CodeGeneration.Tools/Program.cs]: `DESIGN_TOOL_NAME + ".exe"` (spacing)

File: version.props [hunk 4]
```diff
@@ -1,7 +1,7 @@
 <!-- This file may be overwritten by automation. Only values allowed here are VersionPrefix and VersionSuffix.  -->
 <Project>
     <PropertyGroup>
-        <VersionPrefix>2.0.0</VersionPrefix>
+        <VersionPrefix>2.0.1</VersionPrefix>
```
Comment (pranavkm) [inline version.props]: Is this for the patch release?

File: src/Microsoft.VisualStudio.Web.CodeGeneration.Design/Microsoft.VisualStudio.Web.CodeGeneration.Design.csproj [hunk 5]
```diff
@@ -9,14 +9,35 @@
     <OutputType>exe</OutputType>
     <PackageTags>aspnetcore;codegenerator;scaffolding;visualstudioweb</PackageTags>
     <PackageId>Microsoft.VisualStudio.Web.CodeGeneration.Design</PackageId>
-    <RuntimeIdentifier Condition=" '$(TargetFramework)' != 'netcoreapp2.0' ">win7-x86</RuntimeIdentifier>
+    <RuntimeIdentifier Condition=" '$(TargetFramework)' != 'netcoreapp2.0' ">win7-x64</RuntimeIdentifier>
```
Comment (pranavkm) [inline src/Microsoft.VisualStudio.Web.CodeGeneration.Design/Microsoft.VisualStudio.Web.CodeGeneration.Design.csproj]: Looks good. As discussed, you could simplify by this by either using this project do generate rid specific exes or have a common property file so you don't repeat the same details multiple times over.

File: version.props (hunk 4 above)
Comment (prafullbhosale) [inline version.props]: If @Eilon approves 😃

**Comment by jamesabolton** [comment]
Any update on this being released?

**Comment by Eilon** [comment]
@jamesabolton we don't have a precise ETA for the release, but we're setting up a temporary package feed where you'll be able to try out the pre-release builds of the patch. I'll update all the patch-related issues w/ info on the feed when it's available. We're working on getting the 1.0.x and 1.1.x patches ready first, and then the 2.0.x patch.

**Comment by jamesabolton** [comment]
Thank you, @Eilon. As noted in that other bug on dotnet/cli the new projects for Core 2.0 fail with an exception due to this issue. I haven't tested all of them but the MVC and Angular ones do not work. If you remove reference to the DLL then the web UI doesn't function.

**Comment by jamesabolton** [comment]
Is there anything we can do locally to get around this issue to work with the new templates and net standard 2.0 until a patch is released? I tried to downgrade the project that was created to 1.1/1.2 and that breaks various parts of the code in the template project.

**Comment by ddimitrov90** [comment]
any new info about this issue? 
can't really do anything to workaround it and use scaffolding in aspnet core 2.0 in Service Fabric project :(

**Comment by prafullbhosale** [comment]
> Is there anything we can do locally to get around this issue to work with the new templates and net standard 2.0 until a patch is released? I tried to downgrade the project that was created to 1.1/1.2 and that breaks various parts of the code in the template project.

The only possible workaround at the moment is to set the RuntimeIdentifier to `win7-x86` to use scaffolding. Then change it back to the x64 variant.

**Comment by mshindal** [comment]
What about the other Windows RIDs, like `win10-x64`?

**Comment by Eilon** [comment]
@mshindal in general the `win7-XYZ` RIDs will run on newer OSes, including the `win10-XYZ` RIDs. Do you have a specific concern?

**Pull request merged**